homeassistant/components/homematic/cover.py

```diff
@@ -36,6 +39,9 @@ def current_cover_position(self):
 
         None is unknown, 0 is closed, 100 is fully open.
         """
+        if self._hmdevice.__class__.__name__ in HM_GARAGE:
```

Comment:
It feels weird to hack this in like this. 

Instead you should create a new class `HMGarage` and inside `setup_platform` instantiate the right class

Reply:
Yeah, 2x entity class like i.e. Nuki

Reply:
Ok, I misunderstood the original comment by @pvizeli. Will do that, but I can't do anything before next week

Reply:
I've now created the HMGarage class, and I'm using this for the methods that need to be different. Could you take another look?